File: Ruby/rankA_Up/DeterminingTheMap/step2.rb
# 行数 H , 列数 W の盤面があり、各マスには文字が 1 つだけ書かれています。盤面と N 個 の y , x 座標 が与えられるので、盤面の与えられた座標の文字を "#" に書き換えた後の盤面を出力してください。

# なお、マスの座標系は左上端のマスの座標を ( y , x ) = ( 0 , 0 ) とし、
# 下方向が y 座標の正の向き、右方向が x 座標の正の向きとします。

h, w, n = gets.chomp.split(" ").map(&:to_i)
board = []

h.times do
    board << gets.chomp.split("")
end

coordinate = []
n.times do 
    coordinate << gets.chomp.split(" ").map(&:to_i)
end

coordinate.each do |y, x|
    board[y][x] = "#"
end

board.each do |row|
    puts row.join("")
end

File: Python/rankA_Up/EncampmentGame/step2.py
# 盤面の情報が与えられます。現在プレイヤーのいるマスは '*' になっており、そのマスはプレイヤーの陣地です。

# プレイヤーは現在の陣地から上下左右に 1 マス移動することで到達できるマスをプレイヤーの陣地にします。
# ただし、障害物( '#' ) のあるマスは陣地にできません。

# プレイヤーの陣地を '*' にした盤面を出力してください。


H, W = map(int, input().split())
s = [list(input()) for _ in range(H)]

flag_out = False
for y in range(H):
    for x in range(W):
        if s[y][x] == "*":
            if y > 0 and s[y-1][x] != "#":
                s[y-1][x] = "*"
            if y < H - 1 and s[y+1][x] != "#":
                s[y+1][x] = "*"
            if x > 0 and s[y][x-1] != "#":
                s[y][x-1] = "*"
            if x < W - 1 and s[y][x+1] != "#":
                s[y][x+1] = "*"
            flag_out = True
            break
    if flag_out:
        break

for y in range(H):
    print("".join(s[y]))